src/main/java/org/openstreetmap/josm/plugins/pt_assistant/gui/PTAssistantSnackbar.java

```diff
@@ -0,0 +1,94 @@
+// License: GPL. For details, see LICENSE file.
+
+package org.openstreetmap.josm.plugins.pt_assistant.gui;
+
+import static org.openstreetmap.josm.tools.I18n.tr;
+
+import java.awt.Color;
+import java.awt.Cursor;
+import java.awt.Font;
+import java.awt.GridBagLayout;
+import java.util.Timer;
+import java.util.TimerTask;
+import javax.swing.JButton;
+import javax.swing.JPanel;
+import javax.swing.border.CompoundBorder;
+import javax.swing.border.EmptyBorder;
+import javax.swing.border.EtchedBorder;
+
+import org.openstreetmap.josm.gui.MainApplication;
+import org.openstreetmap.josm.gui.widgets.JMultilineLabel;
+import org.openstreetmap.josm.tools.Logging;
+import org.openstreetmap.josm.tools.GBC;
+import org.openstreetmap.josm.gui.MapFrame;
+import org.openstreetmap.josm.tools.ImageProvider;
+
+/**
+ * Snackbars (or useful messages) that are displayed on the bottom of the map
+ *
+ * @author sudhanshu2
+ */
+public class PTAssistantSnackbar extends JPanel {
+
+    /**
+     * Creates a snackbar
+     *
+     * @param message the plain text string to be displayed
+     * @param seconds time the message should be displayed for, set it to -1 for infinite (persist message)
+     */
+    public PTAssistantSnackbar(String message, long seconds) {
+        Font font = getFont().deriveFont(Font.PLAIN, 14.0f);
+        JMultilineLabel snackBarLabel = new JMultilineLabel(tr(message));
+        snackBarLabel.setFont(font);
+        snackBarLabel.setForeground(Color.BLACK);
+
+        JButton closeBtn = new JButton(ImageProvider.get("misc", "black_x"));
+
+        /* set button to transparent, rollover-able, and no border */
+        closeBtn.setContentAreaFilled(false);
+        closeBtn.setRolloverEnabled(true);
+        closeBtn.setBorderPainted(false);
+
+        closeBtn.setCursor(Cursor.getPredefinedCursor(Cursor.HAND_CURSOR));
+        closeBtn.setToolTipText(tr("Close this message"));
+
+        /* This needs to be fixed in order to accommodate for multiple snackbars */
+        closeBtn.addActionListener(e -> {
+            removeSnackbar();
+        });
+
+        setLayout(new GridBagLayout());
+        add(snackBarLabel, GBC.std(1, 1).fill());
+        add(closeBtn, GBC.std(2, 1).span(1, 1).anchor(GBC.EAST));
+        setBorder(new CompoundBorder(new EtchedBorder(EtchedBorder.LOWERED), new EmptyBorder(12, 12, 12, 12)));
+        setBackground(new Color(224, 236, 249));
+
+        MapFrame map = MainApplication.getMap();
```

Comment:
This work should not be done in the constructor.

It is good programming style to only do initialization of the current object or the children in the constructor. But you should not modify the UI by constructing the object. So basically, you should be able to call `new PTAssistantSnackbar()` as often as you want without JOSM changing.

You should move this code to a new method, like this:
```java
snackbar = new PTAssistantSnackbar(initialMessage);
snackbar.addToJosm(seconds);
// ... and then later
snackbar.removeFromJosm();
```
Notice that you could call add/remove multiple times in this variant.